Signal/src/call/UserInterface/CallUIAdapter.swift

```diff
@@ -67,8 +67,8 @@ extension CallUIAdaptee {
         SwiftAssertIsOnMainThread(#function)
 
         guard self.callService.call == nil else {
-            Logger.info("unexpectedly found an existing call when trying to start outgoing call: \(recipientId)")
-            //TODO terminate existing call.
+            owsFail("unexpectedly found an existing call when trying to start outgoing call: \(recipientId)")
+            self.callService.terminateCallDueToCriticalError()
```

Comment:
Same deal: I haven't figured out a repro yet, but I hit this once today.  I got the phone into a state where it couldn't start outgoing calls because of a lingering call.

Reply:
I repro'd again.  You have to terminate a call (locally) then tap call button ASAP.  It's hard to repro, but it's possible.

Reply:
Thinking about this scenario:

1. Alice initiates call to Bob, starts ICE negotiation
2. While Bob/Alice are still setting up the call Signaling (so Bob's phone isn't ringing yet), Bob taps the button to  initiate a call to Charlie.

So before this change:
1. Bob's device would not initiate a call to Charlie, and just seem unresponsive.
2. Then eventually...
2.a. Bob's phone would start to ring with an incoming call from Alice
2.b. or the call would fail to establish and Bob would see a missed call from Alice
3. Now Bob can initiate a call to Charlie.
 
And after this change:
1. the call from Alice would terminate before it has connected
2. Bob would see "Call failed" (for the call from Alice)
3. Bob's call to Charlie would not start.

It's kind of a hairy situation, so I'm not even sure what the ideal behavior for the user is... maybe:

1. When Bob taps to call Charlie
2. Bob's client terminates the call with a "Busy" message (we'd have to ensure all clients handle this: receiving busy *after* a call has started to negotiate ICE might be unexpected).
3. Bob sees a "missed call" notification inserted in his conversation with Alice
4. Bob starts to connect a call with Charlie.

Reply:
I agree with your analysis, but after I opened this PR I found that there are multiple repros.  As you point out, there's the "race to initiate call" case, but there's also a "call service ends up in bad state" case.  In the latter, we want to terminate to get bad to a "known good" state.  In the former, we don't.  

Also, obviously the latter case needs more investigation.  I'm going to revert this part of the PR and see if I can repro the latter case.